### PR TITLE
feat/Add: linting workflow for code quality checks

### PR DESCRIPTION
**Pull request by bupd**
## Fixes: #111 

This pull request introduces a linting workflow to enhance code quality by performing automated checks on our codebase. The added workflow utilizes [Golangci-lint](https://github.com/golangci/golangci-lint) to analyze and enforce coding standards, ensuring consistency and identifying potential issues early in the development process.

### Changes:
- Added .github/workflows/lint.yml to define the linting workflow.
- Configured the workflow to run on every pull request and push to the main branch.
- Integrated [tool name or service] to perform linting checks on our codebase.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  push:
+    paths:
+      - "**.go"
+      - go.mod
+      - go.sum
+  pull_request:
+    paths:
+      - "**.go"
+      - go.mod
+      - go.sum
```
Comment (amands98) [inline .github/workflows/lint.yml:12]:
```suggestion
pull_request:
    branches:
      - main
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
```
Comment (amands98) [inline .github/workflows/lint.yml:18]: This include both golangci and fmt in one job ?

Comment (bupd) [reply .github/workflows/lint.yml:18]: yes

**Comment by bupd** [comment]
@Vad1mo @amands98 ready to merge!.

**Comment by amands98** [comment]
did you test the workflow at your end.

**Comment by bupd** [comment]
Yes, and this is the result.
actRunError-cli

The failure occurs because of some errors being unchecked.

**Review by amands98 — APPROVED** [review]
LGTM

**Comment by Vad1mo** [comment]
we are moving to dagger, we should do it in dagger!

**Pull request closed without merging**